components/clarity-repl/src/analysis/check_checker/mod.rs

```diff
@@ -371,7 +371,9 @@ impl<'a> ASTVisitor<'a> for CheckChecker<'a, '_> {
         // Upon entering a public function, all parameters are tainted
         if let Some(params) = parameters {
             for param in params {
-                self.add_taint_source(Node::Symbol(param.name), param.decl_span);
+                if param.type_expr.match_atom() != Some(&ClarityName::from("bool")) {
+                    self.add_taint_source(Node::Symbol(param.name), param.decl_span)
+                }
```

Comment:
Let's standardize this for more flexibility in the future. Maybe something like this:

```rust
                match TypeSignature::parse_type_repr(param.type_expr, &mut ()) {
                    Ok(TypeSignature::BoolType) => (),
                    Ok(_) => self.add_taint_source(Node::Symbol(param.name), param.decl_span),
                    _ => {
                        // This error will have already been reported by the type-checker. 
                        // Just stop traversing.
                        return false;
                    }
                }
```